tests/test_datasources.py

```diff
@@ -61,6 +61,9 @@ def test_print_data_sources(allkindofcomplexity, server_type):
 
 
 def test_data_sources_from_data_sources(allkindofcomplexity, server_type):
+    with pytest.raises(ValueError) as e:
+        data_sources_false = dpf.core.DataSources(data_sources="Wrong Input", server=server_type)
+        assert "Data source must be gRPC data sources message type" in e
```

Comment:
Minor: if I were you, I would just assert that "gRPC" data sources" substring is contained in the error message.